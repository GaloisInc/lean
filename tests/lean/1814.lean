example : ∀ (p q : Prop), p → q → p :=
begin
  intros p p h1 h2,
  exact h2,
end

example : ∀ (p q : Prop), p → q → p :=
begin
  intros p p h1 h2,
  dedup,
  exact h2,
end
