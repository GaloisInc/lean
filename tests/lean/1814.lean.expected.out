1814.lean:4:8: error: invalid type ascription, term has type
  p
but is expected to have type
  p
types contain aliased name(s): p
remark: the tactic `dedup` can be used to rename aliases
state:
p p : Prop,
h1 : p,
h2 : p
⊢ p
1814.lean:11:8: error: invalid type ascription, term has type
  p_1
but is expected to have type
  p
state:
p p_1 : Prop,
h1 : p,
h2 : p_1
⊢ p
